docs/proposals/08-shoot-apiserver-via-sni.md

```diff
@@ -206,136 +206,102 @@ Another issue occurs when the client tries to talk to the apiserver via the in-c
 
 To mitigate this problem an additional proxy must be deployed on every single Node. It does not terminate TLS and sends the traffic to the correct Shoot API Server. This is achieved by:
 
-- the apiserver master service reconciler is stopped (`--endpoint-reconciler-type=none`).
+- the apiserver master service reconciler is started and pointing to the `kube-apiserver`'s Cluster IP (e.g. `--advertise-address=10.1.2.3`).
 - the proxy runs in the host network of the `Node`.
 - the proxy has a sidecar container which:
-  - creates a dummy network interface and assigns the master service IP address to it.
-  - enables (with iptables) traffic to that IP address (kube-proxy rejects all traffic to `Services` without endpoints) and it removes connection tracking (conntrack) as the IP address is local to the `Node`.
-- the proxy needs to send multiple TLS streams over a single TCP connection (to reduce performance overhead).
-- the proxy listens on the master service IP address and it sends traffic to the correct API server.
+  - creates a dummy network interface and assigns the `10.1.2.3` to it.
+  - removes connection tracking (conntrack) as the IP address is local to the `Node`.
+- the proxy listens on the `10.1.2.3` and using the [PROXY protocol](http://www.haproxy.org/download/2.0/doc/proxy-protocol.txt) it sends the data stream to the Envoy ingress gateway (EIGW).
+- EIGW then sends the TCP stream to another Envoy proxy which listens for PROXY protocol. This proxy then reads the destination IP + port from the PROXY protocol and forwards traffic to that IP + port. As a malicious party could try to exploit and communicate with another component in the Seed cluster, ACL is **mandatory**. Two options are available:
```

Comment:
I have missed why we need two envoy proxies, isn't it possible to customize EIGW to fetch the IP,Port and directly forward to the API-server?

Reply:
The problem is the security aspect. This was the main reason.